bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/discovery/NetworkDiscoveryService.java

```diff
@@ -120,67 +116,117 @@ public void partialDetectionResult(PresenceDetectionValue value) {
     public void finalDetectionResult(PresenceDetectionValue value) {
     }
 
+    private ExecutorService createDiscoveryExecutor(@Nullable NetworkBindingConfiguration configuration) {
+        AtomicInteger count = new AtomicInteger(1);
+        int numThreads = configuration == null ? NetworkBindingConfiguration.DEFAULT_DISCOVERY_THREADS
+                : configuration.numberOfDiscoveryThreads;
+        if (numThreads > 0) {
+            return Executors.newFixedThreadPool(numThreads, r -> {
+                Thread t = new Thread(r, "OH-binding-network-discoveryWorker-" + count.getAndIncrement());
+                t.setDaemon(true);
+                return t;
+            });
+        } else {
+            return Executors.newCachedThreadPool(r -> {
+                Thread t = new Thread(r, "OH-binding-network-discoveryWorker-" + count.getAndIncrement());
+                t.setDaemon(true);
+                return t;
+            });
+        }
+    }
+
     /**
      * Starts the DiscoveryThread for each IP on each interface on the network
      */
     @Override
     protected void startScan() {
-        if (executorService == null) {
-            executorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() * 2);
+        NetworkBindingConfiguration configuration = getConfig();
+        final ExecutorService service;
+        synchronized (this) {
+            if (executorService == null) {
+                executorService = createDiscoveryExecutor(configuration);
```

Comment:
As `executorService` is created only the first time, that means if there is a change of number of threads in the configuration, this will not adjust the thread pool.

Reply:
I now see that it is a binding parameter (and not a thing parameter).
I do not remember, when you change such parameter, the binding is automatically restarted ? If true, I can close my comment ;)

Reply:
i have added it to the documentation.

Reply:
No. The executor service is shut down and nulled in `stopScan()`, so the only time it won't be `null` here is if it's somehow already running a scan.

Reply:
@lsiepel Don't document it - it's wrong. I've changed it and scanned many times during my testing, it works just fine as long as the previous scan had finished.